src/librustdoc/config.rs

```diff
@@ -374,6 +377,31 @@ impl Options {
             }
         };
 
+        let mut default_settings: Vec<Vec<(String, String)>> = vec![
+            matches
+                .opt_str("default-theme")
+                .iter()
+                .map(|theme| {
+                    vec![
+                        ("use-system-theme".to_string(), "false".to_string()),
+                        ("theme".to_string(), theme.to_string()),
+                    ]
+                })
+                .flatten()
+                .collect(),
+            matches
+                .opt_strs("default-setting")
+                .iter()
+                .map(|s| {
+                    let mut kv = s.splitn(2, '=');
+                    let k = kv.next().unwrap().to_string();
+                    let v = kv.next().unwrap_or("true").to_string();
+                    (k, v)
+                })
+                .collect(),
+        ];
+        let default_settings = default_settings.drain(..).flatten().collect();
```

Comment:
This allows reusing the allocation. Not sure if the standard library does that currently, but it could.
```suggestion
        let default_settings = default_settings.into_iter().flatten().collect();
```

Reply:
Thanks.  Perf isn't really important here of course, at least at this level, because this is the command line option parser, but I have no objection to this change.  This change also means getting rid of the `mut` which is actually nicer.  I will add a commit that does both of these things.

Reply:
Right, the perf is not super important to me but I think `into_iter()` is more clear.